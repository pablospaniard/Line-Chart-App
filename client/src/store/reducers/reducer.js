import * as constants from '../constants'

const initialStore = {
  err: null,
  loading: null,
  dataSet: []
}

const dataReducer = (state = initialStore, action) => {
  switch (action.type) {
    case constants.DATA_FETCH_START:
      return { ...state, loading: true }
    case constants.DATA_FETCH_SUCCESS:
      return { ...state, loading: false, dataSet: action.payload.data }
    case constants.DATA_FETCH_FAIL:
      return { ...state, loading: false, err: action.payload }
    default:
      return state
  }
}

export default dataReducer
