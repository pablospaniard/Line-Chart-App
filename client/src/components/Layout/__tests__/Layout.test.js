import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { render, cleanup } from 'react-testing-library'

import reducer from '../../../store/reducers/reducer'
import { Layout } from '..'

describe('Layout should works properly', () => {
  function renderWithRedux(
    ui,
    { initialState, store = createStore(reducer, initialState) } = {}
  ) {
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
      store
    }
  }

  const fetchData = jest.fn()

  const data = {
    recordId: 'test'
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = renderWithRedux(
      <Layout fetchData={fetchData} data={data} />
    )
    expect(container.firstChild).toMatchSnapshot('Layout_snapshot_1')
  })

  it('should show List if loading is false', () => {
    const { getByText } = render(<Layout fetchData={fetchData} data={data} />)
    getByText('Decode')
  })
})
