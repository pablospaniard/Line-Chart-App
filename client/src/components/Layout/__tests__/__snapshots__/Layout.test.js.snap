// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should works properly should match snapshot: Layout_snapshot_1 1`] = `
<div
  class="MuiGrid-container-1 Layout"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      class="MuiGrid-container-1 Header"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        Logo
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
      >
        Your Ambers
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
      >
        Account
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      class="MuiGrid-container-1 Content"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <div
          class="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 Paper"
        >
          <h2
            class="MuiTypography-root-125 MuiTypography-title-131"
          >
            test
          </h2>
          <div
            class="Buttons"
          >
            <button
              class="Button_Select null"
              type="button"
            >
              Decode
            </button>
            <button
              class="null Button_Delete"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
